### ISSUE TITLE
Gaiji

### ISSUE THREAD

**Issue by michaelnmmeyer** [erc-dharma/project-documentation#245]
The usage of `<g>` seems to have changed over time. People typically put a
textual representation of the symbol within the `<g>` element, as in `<g
type="ddanda">||</g>` or even `<g>||</g>`, although the guide says to do things
differently. There are many problems of this type.

**Comment by danbalogh** [comment]
Indeed, there have been changes. There is also encoder inattentiveness. As best I can recall, neither `<g type="ddanda">||</g>` nor `<g>||</g>` was ever permitted by our guidelines, so all of these are mistakes.

For your information, in case this helps with the interpretation of the sad reality you find in the files, here's my reconstruction of former recommendations.

1. `<g type="numeral">` with specific content (mostly numeral characters) has always been permitted for encoding non-decimal numeral symbols, as per EGD §4.2.2. This practice has evolved a bit but there have been no great changes and I think this thread is only about `<g/>` without `@type="numeral"`, so I'm ignoring numerals hereafter.
2. Initially, `<g/>` for symbols (non-alphanumeric glyphs) was always mandatorily an empty element. No content permitted in any circumstances. Then at some point quite early on in the project's lifetime (probably late spring or early summer 2020, but I don't know exactly), we decided to allow content in `<g/>`, strictly limited to the following. The specified content is mandatory in these cases, and no other content is permitted in any case.
- When `<g/>` represents a symbol deemed to be a punctuation mark (EGD §4.2.4), then the encoding must be `<g>.</g>` regardless of the physical appearance or the semantic function or the hierarchical level of the punctuation mark.
- When `<g/>` represents a symbol or symbols deemed to be a space filler (EGD §4.2.5), then the encoding must be `<g>§</g>`, with the § sign iterated as many times as there are distinct symbols in the original (i.e. multiple space fillers are encoded not with multiple iterations of `<g>§</g>`, but rather with `<g>§§§§</g>` and suchlike).
3. The typology of non-alphanumeric symbols has always been something of a problem, as you have noticed. We have vacillated between using a combination of `@type` and `@subtype` versus using just `@type` with more complex tokens, and I know there are instances of both in the corpus. The latter is the officially adopted practice now. I have spent countless hours trying to systematise the symbols occurring in the texts, but ever since the beginning, and increasingly with time, my feeling has been that we are overdoing this, and the complexity of the encoding should be kept to a minimum. We now have several dozen recommended symbol tokens, but every now and then Arlo or someone pops up with a new symbol and wants my opinion on what "species" it should go in and what specific token it should get, since the existing dozens aren't accurate enough. It should probably be me who stands up and says we must put an end to this once and for all, establishing a very small number of tokens (probably headings 1 through 9 plus "miscellaneous" in the [Symbol Taxonomy](https://docs.google.com/document/d/1glfyQnFqPrbVOYzegfjKIOVrc-vMgznEQ1iNsFf7DE8/)), and insist that all other detail be described for human readers in the palaeographic description or the commentary. At any rate, I think there is nothing you need to do in the near future about the type and possible subtype of symbols.

So that's it for history. As for eliminating mistakes in existing files, I think it should be safe to blindly replace all instances of `<g>|</g>`, `<g>||</g>` (two bars) and `<g>‖</g>` (one double bar) to `<g>.</g>`, keeping whatever attributes the containing g element has. Once this is done, `<g>` elements containing any text other than "." or any number of § (including more than two iterations of |) should be flagged for the attention of the encoder who wrote these, who in turn should check and correct manually.

**Comment by arlogriffiths** [comment]
Just on the point of encoder inattentiveness: I think there is a role for both @danbalogh and @michaelnmmeyer to help our encoders reach a higher level of compliance. None of them have the same command of our encoding rules (EGD) as Dan does and none comes close to haveing the same talent for this kind of work. None has the kind of overview of the implications of wrong encoding that Michaël does.

I think it would be good if Michaël good give us some statistics on the corpora with greatest levels of non-compliance and names of the encoders most prone to errors. Then Dan could offer a few training sessions to the culprits, going through a few of their xml files and pointing out corrections that need to be made. Next improving their remaining files will be up to them. Such a pedagogical effort is probably needed on some other fronts as well.

**Comment by danbalogh** [comment]
Doing more coaching is of course OK on my side, but... 
For one thing, I must admit that contrary to the belief of some people, I don't have the EGD in my head. When there is a need to encode (or check the encoding of) anything that I do not frequently deal with in my own texts, all I can do is turn to the EGD and look up the correct way to do it - which is something anybody can do without my assistance. It may take a bit less time for me to find the right "chapter and verse", but not an order of magnitude less.
For another, you (Arlo) and I have recently discussed something much like this apropos of Michaël's list of files with schema conflicts. I don't think Michaël could give us statistics of "levels of non-compliance" beyond schema conflicts (unless we ask individually for specific kinds of errors that do not conflict the schema, like `<g type="ddanda">||</g>`), and he has already done that much. Back then, I corrected my own errors and went and looked at some of the files encoded by others and flagged by Michaël. I then sent out a circular (3 Nov) to everyone in the project, listing the most common kinds of mistakes that seem to occur and offering to help with questions - which nobody responded to, though for all I know, some or all of those schema conflicts may have been corrected in the meantime by the respective encoders. Singling out individual "culprits" for sessions in which I play schoolmaster with them may - possibly - be more effective than sending out circulars and hoping people will act on them, but is certainly not an efficient use of anyone's time.

**Comment by michaelnmmeyer** [comment]
@danbalogh Thank you for these precisions. The schema did not allow multiple occurrences of § in `<g>`, this is fixed in 9f09520.

@arlogriffiths It is not really feasible to extract useful metrics, because there is no typology of schema-related errors: generally speaking, it is not possible to tell if someone makes 100 unrelated mistakes or makes the same mistake 100 times.

**Comment by arlogriffiths** [comment]
@michaelnmmeyer — I realized after signing off last night that a group of offending files is probably those imported from the earlier Campā corpus hosted as https://isaw.nyu.edu/publications/inscriptions/campa/index.html. I am sure the encoding of `<g>` was different in the files I encoded in 2012 and has not yet been brought in line with DHARMA EGD in the batch import carried out by Axelle. Let me work on that over the coming weeks, maybe with @salomepichon's help, and then we'll see if the number of offending cases has become significantly reduced. I agree with @danbalogh that it should be OK "to blindly replace all instances of `<g>|</g>`, `<g>||</g>` (two bars) and `<g>‖</g>` (one double bar) to `<g>.</g>`, keeping whatever attributes the containing g element has" (but hope Dan meant that the nature of the symbol previously enclosed in `<g>` should determine what attribute to assign to the new `<g>`, e.g. @type="ddanda" if it was ||). I also agree that, "once this is done, `<g>` elements containing any text other than "." or any number of § (including more than two iterations of |) should be flagged for the attention of the encoder who wrote these, who in turn should check and correct manually."

As for your remark about pedagogy, @danbalogh, I think we need more of it, for individual team members or small groups. This is part of your job. Sending group message is clearly not effective for anything except saving time. I think even without statistics from Michaël, you could approach team members individually and propose a 2-hour zoom session with each individual over the next weeks and months, to go through one or more files they have encoded, to see what problems emerge, and determine whether firther coaching is necessary. Please discuss with @AnneSchmiedchen.

As for the Symbol Taxonomy guide, I think thanks to your effort it has become quite a useful documents, and I would urge it to be prepared for formal deposit in HAL (though there is no great hurry). If you feel our approach must be drastically simplified, then I can also live with that, though I'd prefer to sgtick to a level of granularity close to what we have in place now, and that has generally been working well for inscriptions I have been encoding over the past years.

**Comment by danbalogh** [comment]
Re blind replacement: I wasn't expressing myself clearly enough. What I had in mind was cases like `<g type="ddanda">||</g>`, where the attribute already present on the `<g>` should be retained. So the `<g>` in my examples stood for "instances of g elements with any attributes". I had overlooked that there were also cases of g without attributes but containing vertical bars. So the blind replaces should be more fine-tuned, as follows:
```
<g> without any attributes, containing | (single vertical bar) --- to <g type="danda">.</g>
<g> without any attributes, containing || (two vertical bars) or ‖ (double vertical bar) --- to <g type="ddanda">.</g>
<g> with any attributes, containing | (single vertical bar) --- to <g type="danda">.</g> preserving the attributes present on the g
<g> with any attributes, containing || (two vertical bars) or ‖ (double vertical bar) --- to <g type="ddanda">.</g> preserving the attribute present on the g
```

Re pedagogy: fair enough, if you feel it really is needed then I am happy to do it. I'll talk to Annette about it and start scheduling something from January onward.

Re Symbol Taxonomy, we should also pick this up sometime early next year when you have time. Before formally publishing anything, I think we should, with Michaël's help, do what has been the intent from the start and harvest all instances that we now have in the encoded corpus, which I would then look at and think about. Then we should probably revert to using type and subtype, with a small controlled vocabulary of available types and an open-ended list of subtypes for each type if you really are certain we need those.

**Comment by michaelnmmeyer** [comment]
No problem for fixing `<g>|</g>` and so on, I will add a transformation rule for that.

One remark on the pedagogy thing. Axelle and I noticed that people sometimes do not know how to interpret validation error messages. I can assist them with that, if they are willing to "debug" their files by addressing errors methodically and systematically. This is not a particularly fun activity, and we still have a lot of time, so it might be more useful to do this once or twice, say, 1 year before the end of the project.

**Comment by arlogriffiths** [comment]
I think any such efforts of pedagogy are welcome, both for issues of non compliance with EGD that do and that do not provoke validation errors. (There are many more of the latter than of the former.)

I suggest that @danbalogh and @michaelnmmeyer each separately take up contact with small groups of encoders and propose zoom sessions during which Michael could focus on fostering a habit of paying atternton to validation errors and show how to debug them, while Dan during his sessions could focus on tricky aspects of encoding like representing intrinsic and extrinsic structure.

**Comment by danbalogh** [comment]
Will do.

**Comment by michaelnmmeyer** [comment]
As well.
